lib/puppet/provider/service/systemd.rb

```diff
@@ -7,6 +7,7 @@
 
   defaultfor :osfamily => [:archlinux]
   defaultfor :osfamily => :redhat, :operatingsystemmajrelease => "7"
+  defaultfor :osfamily => :redhat, :operatingsystem => :fedora, :operatingsystemmajrelease => ["17", "18", "19", "20", "21"]
```

Comment:
On that same note, I'd certainly prefer something like:

```ruby
defaultfor do
  Facter.value(:operatingsystem) == 'Fedora' && Facter.value(:operatingsystemmajrelease).to_i >= 17
end
```

Unfortunately `defaultfor` isn't expressive enough to enable this as it only allows for feature or fact value comparisons.  We could certainly extend it to support blocks, though.

The reason I prefer it this way is that we would only have to fix the provider when Fedora moves away from systemd, which I'm sure will happen less frequently than Fedora doing a release.